### PR TITLE
Add collapse prompt buffer command

### LINKED ISSUE #2788: Collapse prompt buffer when focused out
**Is your feature request related to a problem? Please describe.**
There's often a use pattern of prompt buffer, where a person opening a prompt buffer focuses the main buffer behins it and scrolls it some. The prompt buffer suggestions seem useless for this person, as their attention is on the main buffer.

**Describe the solution you'd like**
How about we collapse the prompt buffer to the input size when it's focused out? This way, prompt buffer is still there, but takes much less screen real estate.

**Additional context**
https://github.com/atlas-engineer/nyxt/pull/2774#discussion_r1102717455

### PR DESCRIPTION
**Pull request by aadcg**
# Description

Fixes #2788.

# Discussion

I'm not a huge fan of the keybinding I've choose. Suggestions are welcome.

Find a little demo below (@lansingthomas, I did it for you). After all these years, there are still little and trivial UI improvements that make a huge difference. 

https://github.com/atlas-engineer/nyxt/assets/45483512/f57c4846-1317-4b2e-9465-a124f71578ec

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.

- [x] Git hygiene:
  - I have pulled from master before submitting this PR
  - There are no merge conflicts.
- [x] I've added the new dependencies as:
  - ASDF dependencies,
  - Git submodules,
    ```sh
	cd /path/to/nyxt/checkout
    git submodule add https://gitlab.common-lisp.net/nyxt/py-configparser _build/py-configparser
    ```
  - and Guix dependencies.
- [x] My code follows the style guidelines for Common Lisp code. See:
  - [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf)
  - [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml)
- [x] I have performed a self-review of my own code.
- [ ] My code has been reviewed by at least one peer.  (The peer review to approve a PR counts.  The reviewer must download and test the code.)
- [ ] Documentation:
  - All my code has docstrings and `:documentation`s written in the aforementioned style.  (It's OK to skip the docstring for really trivial parts.)
  - I have updated the existing documentation to match my changes.
  - I have commented my code in hard-to-understand areas.
  - I have updated the `changelog.lisp` with my changes if it's anything user-facing (new features, important bug fix, compatibility breakage).
  - I have added a `migration.lisp` entry for all compatibility-breaking changes.
  - (If this changes something about the features showcased on Nyxt website) I have these changes described in the new/existing article at Nyxt website or will notify one of maintainters to do so.
- [ ] Compilation and tests:
  - My changes generate no new warnings.
  - I have added tests that prove my fix is effective or that my feature works.  (If possible.)
  - I ran the tests locally (`(asdf:test-system :nyxt)` and `(asdf:test-system :nyxt/gi-gtk)`) and they pass.

### CODE REVIEW

**Review by aartaka — APPROVED** [review]
Yeah, I'm not really into this binding either, but the logic looks right at least :)

File: source/mode/prompt-buffer.lisp [hunk 1]
```diff
@@ -316,6 +317,12 @@ current unmarked suggestion."
           attributes)
     (prompt-render-suggestions prompt-buffer)))
 
+(define-command-prompt toggle-suggestions-display (prompt-buffer)
+  "Toggle between showing suggestions."
+  (if (eq (height prompt-buffer) :default)
+      (setf (height prompt-buffer) :fit-to-prompt)
+      (setf (height prompt-buffer) :default)))
```
Comment (Ambrevar) [inline source/mode/prompt-buffer.lisp:324]: A bit more Lispy: move the condition to the value of the `setf`.

Comment (aadcg) [reply source/mode/prompt-buffer.lisp:324]: @Ambrevar are you suggesting to write it as (see below)?

```lisp
(let ((current-height (height prompt-buffer)))
    (setf (height prompt-buffer)
          (if (eq current-height :fit-to-prompt) :default :fit-to-prompt)))
```

Comment (Ambrevar) [reply source/mode/prompt-buffer.lisp]: Yes!

**Comment by aadcg** [comment]
@Ambrevar ok to merge? Thanks.

**Comment by Ambrevar** [comment]
OK! :)

**Comment by aadcg** [comment]
Ups, I've forgot the changelog entry. Added in 8dd831b12.

**Pull request merged**